homeassistant/components/kmtronic/switch.py

```diff
@@ -0,0 +1,88 @@
+"""KMtronic Switch integration."""
+from datetime import timedelta
+import logging
+
+import aiohttp
+import async_timeout
+
+from homeassistant.components.switch import SwitchEntity
+from homeassistant.helpers.update_coordinator import (
+    CoordinatorEntity,
+    DataUpdateCoordinator,
+    UpdateFailed,
+)
+
+from .const import DATA_HOST, DATA_HUB, DOMAIN, MANUFACTURER
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass, entry, async_add_entities):
+    """Config entry example."""
+    hub = hass.data[DOMAIN][entry.entry_id][DATA_HUB]
+    host = hass.data[DOMAIN][entry.entry_id][DATA_HOST]
+    await hub.async_get_relays()
+
+    async def async_update_data():
+        try:
+            async with async_timeout.timeout(10):
+                await hub.async_update_relays()
+        except aiohttp.client_exceptions.ClientConnectorError as err:
+            raise UpdateFailed(f"Error communicating with API: {err}") from err
+
+    coordinator = DataUpdateCoordinator(
```

Comment:
Can we set this up in `__init__` and raise `ConfigEntryNotReady` if the device isn't reachable so its retried later?

Also if the auth credentials have changed we should log an error and return False

In a future PR implement https://developers.home-assistant.io/docs/config_entries_config_flow_handler/#reauthentication